extensions/publisher-command-center/src/components/UnauthorizedView.js

```diff
@@ -0,0 +1,94 @@
+import m from "mithril";
+
+const UnauthorizedView = {
+  oninit: function(vnode) {
+    vnode.state.integration = null;
+    vnode.state.loading = true;
+
+    // Check for available integration
+    m.request({ method: "GET", url: "api/integrations" })
+      .then(response => {
+        vnode.state.integration = response;
+        vnode.state.loading = false;
+        m.redraw();
+      })
+      .catch(err => {
+        console.error("Failed to fetch integrations", err);
+        vnode.state.loading = false;
+        m.redraw();
+      });
+  },
+
+  addIntegration: function(vnode) {
+    if (!vnode.state.integration) return;
+
+    m.request({
+      method: "PUT",
+      url: "api/visitor-auth",
+      body: { integration_guid: vnode.state.integration.guid }
+    })
+      .then(() => {
+        // Reload the top-most window to check authorization again
+        window.top.location.reload();
+      })
+      .catch(err => {
+        console.error("Failed to add integration", err);
+      });
+  },
+
+  view: function(vnode) {
+    // Show loading state
+    if (vnode.state.loading) {
+      return m("div.d-flex.justify-content-center", { style: { margin: "3rem" } }, [
+        m("div.spinner-border.text-primary", { role: "status" },
+          m("span.visually-hidden", "Loading...")
+        )
+      ]);
+    }
+
+    // We have an integration ready to add
+    if (vnode.state.integration) {
+      return m("div.alert.alert-info", { style: { margin: "1rem auto", maxWidth: "800px" } }, [
+        m("div", { style: { marginBottom: "1rem" } }, [
+          m.trust("This content uses a <strong>Visitor API Key</strong> " +
+            "integration to show users the content they have access to. " +
+            "A compatible integration is displayed below." +
+            "<br><br>" +
+            "For more information, see " +
+            "<a href='https://docs.posit.co/connect/user/oauth-integrations/#obtaining-a-visitor-api-key' " +
+            "target='_blank'>documentation on Visitor API Key integrations</a>.")
+        ]),
```

Comment:
Use the `m` component constructor here. You can use this to translate the raw HTML to Mithril code: https://arthurclemens.github.io/mithril-template-converter/